### DISCUSSION TITLE
Add margin to the charging logic when overheating

### DISCUSSION THREAD

**Discussion by devklepacki** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1015] [Ideas]
I noticed a thing when a battery is exceeding a set temperature limit.

1. When the limit is set to 35°C, at 35.2°C charging stops, so the temperature can drop to 35°. This results in Aldente enabling charging for 3 seconds, the temperature again going to 35.2° and the cycle going on.

2. Maybe it would be faster/better/whatever to allow some margin of 1 or 2°C, so e.g. once the set limit is exceeded at 35.2° then Aldente let the battery drop to 34° and only then enable charging again?

This'd probably need to be customizable, because from what I noticed the first scenario actually still allows the battery to be very slowly charged. While the second (with the new margin) could cause the battery to never charge due to the temperature for whatever reason never dropping to 34°.

That's just a thought that'd require a little testing. Maybe implementing such a margin would result in faster charging on average.

**Comment by MatthiasKerbl** [comment]
Hi @devklepacki,

excellent idea as well. There is already a feature request open for this #675, therefore I will close the discussion here.

Best,
Matthias

**Reply by devklepacki** [reply]
Thank you! I searched for this but now I see I was searching for the wrong keywords!
